contrib/gin-gonic/gin/appsec.go

```diff
@@ -9,12 +9,56 @@ import (
 	"net/http"
 
 	"github.com/DataDog/dd-trace-go/v2/appsec"
+	"github.com/DataDog/dd-trace-go/v2/appsec/events"
+	"github.com/DataDog/dd-trace-go/v2/instrumentation/appsec/dyngo"
 	"github.com/DataDog/dd-trace-go/v2/instrumentation/appsec/emitter/httpsec"
 	"github.com/DataDog/dd-trace-go/v2/instrumentation/appsec/trace"
 
 	"github.com/gin-gonic/gin"
+	"github.com/gin-gonic/gin/binding"
 )
 
+// AppsecBinding wraps a binding.BindingBody to add AppSec monitoring of the parsed request body.
+// It is used to override the default bindings in the gin binding package at init time.
+// Keep in mind that this does not cover all the ways to bind a request in gin because of the
+// [binding.BindingBody.BindBody] method that we do not wrap because we would be missing the request context.
+// You can also you it manually by wrapping any binding.BindingBody you want and using it with [gin.Context.MustBindWith]
+// or [gin.Context.BindWith].
```

Comment:
```suggestion
// AppsecBinding wraps a [binding.BindingBody] to add AppSec monitoring of the parsed request body.
// It is used to override the default bindings in the gin binding package at init time.
// Keep in mind that this does not cover all the ways to bind a request in gin because of the
// [binding.BindingBody.BindBody] method that we do not wrap because we would be missing the request context.
// You can also you it manually by wrapping any [binding.BindingBody] you want and using it with [gin.Context.MustBindWith]
// or [gin.Context.BindWith].
```

?